### PR TITLE
docs(*): DPP lifecycle for universal and k8s

### PR DESCRIPTION
**Pull request by lobkovilya**
Fix https://github.com/kumahq/kuma-website/issues/606

### CODE REVIEW

File: docs/docs/dev/explore/dpp-on-universal.md [hunk 1]
```diff
@@ -66,4 +66,87 @@ For this to work, we must update our application to consume `redis` on `127.0.0.
...
+
+On Universal there are 2 ways how Dataplane resource lifecycle could be managed: [Direct](#direct) and [Indirect](#indirect). 
+
+### Direct
+
+This is a recommended way to operate with Dataplane resources on Universal.
+
+#### Creation
+
+Pass Dataplane resource directly to `kuma-dp run` command. Dataplane resource could be a [Go Template](https://pkg.go.dev/text/template) in this case:
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply docs/docs/dev/explore/dpp-on-universal.md:81]: Right, I was thinking it's just different implementations but same syntax, but apparently it's not

File: docs/docs/dev/explore/dpp-on-universal.md [hunk 2]
```diff
@@ -66,4 +66,87 @@ For this to work, we must update our application to consume `redis` on `127.0.0.
...
+
+Regardless the way data plane proxy went down (gracefull or ungracefull), Dataplane resource won't be cleaned up immediately.
+
+If data plane proxy is shutdown gracefully, then Dataplane resource will be marked as Offline. Offline data plane proxies 
+deleted automatically after `KUMA_RUNTIME_UNIVERSAL_DATAPLANE_CLEANUP_AGE`, by default it's 72h.
+
+If data plane proxy went down ungracefully, then the following sequence of the events should happen:
+1. Since xDS connection between proxy and kuma-cp is inactive, Dataplane proxy will be marked as Offline after `KUMA_METRICS_DATAPLANE_IDLE_TIMEOUT`,
+   by default it happens after 5 minutes.
+2. Offline data plane proxies deleted automatically after `KUMA_RUNTIME_UNIVERSAL_DATAPLANE_CLEANUP_AGE`, by default it's 72h.
```
[the remark this answers is not in the document]
Comment (lobkovilya) [reply docs/docs/dev/explore/dpp-on-universal.md]: Maybe some parameters that needs to be tuned should be propagated from Global CP

File: docs/docs/dev/explore/dpp-on-universal.md [hunk 3]
```diff
@@ -66,4 +66,87 @@ For this to work, we must update our application to consume `redis` on `127.0.0.
...
+
+If data plane proxy is shutdown gracefully, then Dataplane resource will be automatically deleted by kuma-cp. 
+
+If data plane proxy went down ungracefully, then Dataplane resource won't be deleted immediately. The following sequence 
+of the events should happen:
+1. Since xDS connection between proxy and kuma-cp is inactive, Dataplane proxy will be marked as Offline after `KUMA_METRICS_DATAPLANE_IDLE_TIMEOUT`, 
+by default it happens after 5 minutes.
+2. Offline data plane proxies are deleted automatically after `KUMA_RUNTIME_UNIVERSAL_DATAPLANE_CLEANUP_AGE`, by default it's 72h.
+
+It's guaranteed that Dataplane resource will be eventually cleaned up in case of **ungraceful** shutdown.
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp-on-universal.md:119]:
```suggestion
This guarantees that `Dataplane` resources are eventually cleaned up even in the case of **ungraceful** shutdown. 
```

File: docs/docs/dev/explore/dpp-on-universal.md [hunk 4]
```diff
@@ -66,4 +66,87 @@ For this to work, we must update our application to consume `redis` on `127.0.0.
...
+1. Since xDS connection between proxy and kuma-cp is inactive, Dataplane proxy will be marked as Offline after `KUMA_METRICS_DATAPLANE_IDLE_TIMEOUT`, 
+by default it happens after 5 minutes.
+2. Offline data plane proxies are deleted automatically after `KUMA_RUNTIME_UNIVERSAL_DATAPLANE_CLEANUP_AGE`, by default it's 72h.
+
+It's guaranteed that Dataplane resource will be eventually cleaned up in case of **ungraceful** shutdown. 
+
+### Indirect
+
+The lifecycle is called "Indirect", because there is no strict dependency between Dataplane resource creation and the 
+startup of the data plane proxy. That's an old but still valid way to manage Dataplane resources.
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp-on-universal.md:124]: I don't think it should qualified as old.

I think we should just say that this works if you have something that manages lifecycle. For example you'd do this with `Nomad`

File: docs/docs/dev/explore/dpp-on-universal.md [hunk 5]
```diff
@@ -66,4 +66,87 @@ For this to work, we must update our application to consume `redis` on `127.0.0.
...
+startup of the data plane proxy. That's an old but still valid way to manage Dataplane resources. 
+
+#### Creation
+
+Dataplane resource is created using [HTTP API](../reference/http-api.md#dataplanes) or [kumactl](../explore/cli.md). 
+Dataplane resource is created before data plane proxy started. There is no support for templates, resource should be
+a valid Dataplane configuration. 
+
+When data plane proxy is started, it takes `name` and `mesh` as an input arguments. After connection between proxy and 
+kuma-cp is established, kuma-cp will attempt to find Dataplane resource with `name` and `mesh` in the store.
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp-on-universal.md:133]:
```suggestion
kuma-cp is established, kuma-cp finds the `Dataplane` resource with `name` and `mesh` in the store. 
```

There's no try :)

File: docs/docs/dev/explore/dpp-on-universal.md [hunk 6]
```diff
@@ -66,4 +66,87 @@ For this to work, we must update our application to consume `redis` on `127.0.0.
...
+
+Regardless the way data plane proxy went down (gracefull or ungracefull), Dataplane resource won't be cleaned up immediately.
+
+If data plane proxy is shutdown gracefully, then Dataplane resource will be marked as Offline. Offline data plane proxies 
+deleted automatically after `KUMA_RUNTIME_UNIVERSAL_DATAPLANE_CLEANUP_AGE`, by default it's 72h.
+
+If data plane proxy went down ungracefully, then the following sequence of the events should happen:
+1. Since xDS connection between proxy and kuma-cp is inactive, Dataplane proxy will be marked as Offline after `KUMA_METRICS_DATAPLANE_IDLE_TIMEOUT`,
+   by default it happens after 5 minutes.
+2. Offline data plane proxies are deleted automatically after `KUMA_RUNTIME_UNIVERSAL_DATAPLANE_CLEANUP_AGE`, by default it's 72h.
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp-on-universal.md:152]: Same rewrite as before.

**Review by lahabana — COMMENTED** [review]
Thx! That's going to bring a lot of clarity to users!

File: docs/docs/dev/explore/dpp-on-universal.md (hunk 5 above)
Comment (lobkovilya) [reply docs/docs/dev/explore/dpp-on-universal.md:133]: I had in mind that kuma-cp will back off and try again if it doesn't find `Dataplane` with the first attempt, but I didn't make it clear. I don't mind using your option

**Review by lahabana — APPROVED** [review]
Perfect thx!

**Pull request merged**